docs/reference/troubleshooting/common-issues/mapping-explosion.asciidoc

```diff
@@ -0,0 +1,120 @@
+[[mapping-explosion]]
+=== Mapping explosion
+
+{es}'s search and {kibana-ref}/discover.html[{kib}'s discover] Javascript rendering are 
+dependent on the search's backing indices total amount of 
+<<mapping-types,mapped fields>>, of all mapping depths. When this total 
+amount is too high or is exponentially climbing, we refer to it as 
+experiencing mapping explosion. Field counts going this high are uncommon 
+and usually suggest an upstream document formatting issue as 
+link:https://www.elastic.co/blog/found-crash-elasticsearch#mapping-explosion[shown in this blog]. 
+
+Mapping explosion may surface as the following performance symptoms:
+
+* <<cat-nodes,CAT nodes>> reporting high heap or CPU on the main node 
+and/or nodes hosting the indices shards. This may potentially 
+escalate to temporary node unresponsiveness and/or main overwhelm.
+
+* <<cat-tasks,CAT tasks>> reporting long search durations only related to 
+this index or indices, even on simple searches. 
+
+* <<cat-tasks,CAT tasks>> reporting long index durations only related to 
+this index or indices. This usually relates to <<cluster-pending,pending tasks>> 
+reporting that the coordinating node is waiting for all other nodes to 
+confirm they are on mapping update request.
+
+* Discover's **Fields for wildcard** page-loading API command or {kibana-ref}/console-kibana.html[Dev Tools] page-refreshing Autocomplete API commands are taking a long time (more than 10 seconds) or 
+timing out in the browser's Developer Tools Network tab.
+
+* Discover's **Available fields** taking a long time to compile Javascript in the browser's Developer Tools Performance tab. This may potentially escalate to temporary browser page unresponsiveness.
+
+* Kibana's {kibana-ref}/alerting-getting-started.html[alerting] or {security-guide}/detection-engine-overview.html[security rules] may error `The content length (X) is bigger than the maximum allowed string (Y)` where `X` is attempted payload and `Y` is {kib}'s {kibana-ref}/settings.html#server-maxPayload[`server-maxPayload`]. 
+
+* Long {es} start-up durations. See link:https://github.com/elastic/elasticsearch/issues/83203[this discussion].
```

Comment:
```suggestion
* Long {es} start-up durations. 
```

Reply:
I know the Elasticsearch repo is open so anyone can look up issues, but I haven't yet seen us link to GitHub issues in our docs. Admittedly I've spent more time in the Cloud repo and haven't read all the docs yet :).

Reply:
I know Github-references are infrequent. The other doc PR I just did, they asked for the Github reference to be taken out as well, so good to do here. Cheers!